201-web-app-public-certificate/azuredeploy.json

```diff
@@ -0,0 +1,38 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "exisitngWebAppName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the web app that would use this public certificate. The web app should belong to a dedicated App Service Plan."
+            }
+        },
+        "publicCertificateLocation": {
+            "type": "string",
+            "allowedValues": ["CurrentUserMy", "LocalMachineMy"],
+            "metadata": {
+                "description": "Location where you wish to install the public certificate. 'LocalMachineMy' is supported in App Service Environment only."
+            }
+        },
+        "blob": {
+            "type": "string",
+            "metadata": {
+                "description": "Base 64 encoded public certificate file"
+            }
+        }
+    },
+    "variables": {
+        "publicCertificateName": "[concat(uniqueString(resourceGroup().id, deployment().name),'publicCertificate')]"
```

Comment:
Deployment names don't have to be unique but in any case it seems unlikely that the user would want a random name for a resource - the list would become unmanageable since there's no way to associate the name with it's content.  There is specific content behind that name, so the name should be deterministic.